### ISSUE TITLE
Update to use Jakarta Transaction API

### ISSUE THREAD

**Issue by hantsy** [micronaut-projects/micronaut-data#2264]
### Expected Behavior

Should use Jakarta Transaction API instead.

### Actual Behaviour

Micronaut Data still use legacy javax.transaction-api.

### Steps To Reproduce

The result is generated by `./gradlew dependencies`

|    |    |    +--- io.micronaut.data:micronaut-data-tx:4.0.0-M7
|    |    |    |    +--- io.micronaut:micronaut-inject:4.0.0-M6 (*)
|    |    |    |    +--- io.micronaut:micronaut-aop:4.0.0-M6 (*)
|    |    |    |    +--- javax.transaction:javax.transaction-api:1.3
|    |    |    |    \--- io.micronaut:micronaut-core-bom:4.0.0-M6 (*)

### Environment Information

Windows 10, JDK 17

### Example Application

https://github.com/hantsy/micronaut-sandbox/tree/master/data-jpa-post-service

### Version

4.0.0-M3

**Comment by graemerocher** [comment]
@dstepanov @radovanradic seems important? Thought this was done already?

**Comment by radovanradic** [comment]
From Micronaut Data guide https://micronaut-projects.github.io/micronaut-data/latest/guide/#sqlAnnotations
```
Micronaut Data supports both javax.persistence and jakarta.persistence packages.
```
Also from here https://micronaut-projects.github.io/micronaut-data/latest/guide/#jdbcQuickStart
```
If you want to use JPA annotations in your entities with Micronaut Data JDBC, we strongly recommend you use jakarta.persistence annotations. Micronaut Data will remove support for javax.persistence annotations in the future.
```
So we still have javax.persistence dependency. Do we want to completely remove it now in 4.0 ?

**Comment by graemerocher** [comment]
yes needs to be removed and replaced with `jakarta`

**Comment by radovanradic** [comment]
Resolved via #2265
